include/pybind11/pybind11.h

```diff
@@ -2751,10 +2751,15 @@ get_type_override(const void *this_ptr, const type_info *this_type, const char *
         if ((std::string) str(f_code->co_name) == name && f_code->co_argcount > 0) {
             PyObject *locals = PyEval_GetLocals();
             if (locals != nullptr) {
+#        if PY_VERSION_HEX >= 0x030b0000
+                PyObject *co_varnames = PyCode_GetVarnames(f_code);
+#        else
                 PyObject *co_varnames = PyObject_GetAttrString((PyObject *) f_code, "co_varnames");
+#        endif
                 PyObject *self_arg = PyTuple_GET_ITEM(co_varnames, 0);
                 Py_DECREF(co_varnames);
                 PyObject *self_caller = dict_getitem(locals, self_arg);
+                Py_DECREF(locals);
```

Comment:
Oh ... wow.

Could you please explain a little more in the PR description?

What was the motivation/trigger to start this PR? (might be useful context for search engines to pick up on)

And how did you discover the refcount bug? (a tool? or did you just notice while working on the other change here?)

It's not terribly important, but sometimes these things give clues later.